File: remove-duplicates-from-sorted-list.js
// Given the head of a sorted linked list,
// delete all duplicates such that each element appears only once. Return the linked list sorted as well.
var deleteDuplicates = function(head) {
  let currNode = head;
  
  while(currNode !== null && currNode.next !== null){
    if(currNode.val === currNode.next.val){
      currNode.next = currNode.next.next
    }
    else {
      currNode = currNode.next
    }
  }
  
  return head;
  };

File: remove-duplicates-from-sorted-list.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @return {ListNode}
def delete_duplicates(head)
  currNode = head

  while currNode && currNode.next
    if currNode.value == currNode.next.value
      currNode.next = currNode.next.next
    else
      currNode =  currNode.next
    end
  end
    return head
end